user/getting-started.md

```diff
@@ -4,38 +4,107 @@ layout: en
 
 1. [Sign in to Travis CI](https://travis-ci.org/auth) with your GitHub account, accepting the GitHub [access permissions confirmation](/user/github-oauth-scopes).
 
-2. Once you're signed in, and we've synchronized your repositories from GitHub, go to your [profile page](https://travis-ci.org/profile) and enable Travis CI for the repository you want to build.
+2. Once you're signed in, and we've synchronized your GitHub repositories, go to your [profile page](https://travis-ci.org/profile) and enable Travis CI for the *open source* repository you want to build. If you want to build a private repository, sign in to [Travis CI for private repositories](https://travis-ci.com/profile) instead.
 
-   > Note: You can only enable Travis CI builds for repositories you have admin access to.
+3. Add a `.travis.yml` file to your repository to tell Travis CI what to build.
 
-3. Add a `.travis.yml` file to your repository to tell Travis CI what to build:
+   This example tells Travis CI that this is a Ruby project, so unless you change the default, Travis CI uses `rake` to build it.
+
+   Travis CI tests this project against Ruby 2.2 and the latest versions of JRuby and Rubinius, which can all pass or fail independently.
 
    ```yaml
    language: ruby
    rvm:
     - 2.2
     - jruby
     - rbx-2
-   # uncomment and edit the following line if your project needs to run something other than `rake`:
-   # script: bundle exec rspec spec
    ```
 
-   This example tells Travis CI that this is a project written in Ruby and built with `rake`. Travis CI tests this project against Ruby 2.2 and the latest versions of JRuby and Rubinius.
-
 4. Add the `.travis.yml` file to git, commit and push, to trigger a Travis CI build:
 
-   > Travis only runs a build on the commits you push **after** adding the repository to Travis.
+   > Travis only runs builds on the commits you push *after* you've enabled the repository in Travis CI.
+
+5. Check the [build status](https://travis-ci.org/repositories) page to see if your build passes or fails, according to the return status of the build command.
+
+## Selecting a programming language
+
+Use one of these common languages:
+
+```yaml
+language: ruby
+language: java
+language: node_js
+language: android
+language: php
+```
+
+Or pick one from the [full list](/user/languages/).
+
+## Selecting infrastructure
+
+The most straightforward way to determine what infrastructure your build runs on
+is to set the `language`. If you do this your build runs on the default
+infrastructure (with a few exceptions), which is Container Based Ubuntu 12.04.
+You can do this explicitly by adding `sudo: false` to your `.travis.yml`.
+
+* If you need a more up-to-date version of Ubuntu on the same infrastructure, use
+the beta of Ubuntu Linux Trusty 14.04:
+
+   ```yaml
+   sudo: false
+   dist: trusty
+   ```
+
+* If you need a more customizable, fully virtualized environment, use the Sudo
```

Comment:
I am not sure about having this discussion here. Selection of infrastructure should not be a concern for the very first build.